### PR TITLE
[react-dom] Roots accept ReactNode as initialChildren

### PR DESCRIPTION
**Pull request by eps1lon**
Not sure why I thought it's just `ReactChild`.

### CODE REVIEW

File: types/react-dom/client.d.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export interface RootOptions {
 }
 
 export interface Root {
-    render(children: React.ReactChild | Iterable<React.ReactNode>): void;
+    render(children: React.ReactNode): void;
```
Comment (eps1lon) [inline types/react-dom/client.d.ts:25]: https://github.com/facebook/react/blob/fe6e0741286345edb2aa23784c21f8ea611ebdea/packages/react-dom/src/client/ReactDOMRoot.js#L93

File: types/react-dom/client.d.ts [hunk 2]
```diff
@@ -35,6 +35,6 @@ export function createRoot(container: Element | DocumentFragment, options?: Root
 
 export function hydrateRoot(
     container: Element | Document,
-    initialChildren: React.ReactChild | Iterable<React.ReactNode>,
+    initialChildren: React.ReactNode,
```
Comment (eps1lon) [inline types/react-dom/client.d.ts:38]: https://github.com/facebook/react/blob/fe6e0741286345edb2aa23784c21f8ea611ebdea/packages/react-dom/src/client/ReactDOMRoot.js#L257
`ReactNodeList` is our `ReactNode`

File: types/react-dom/test/react-dom-tests.tsx [hunk 3]
```diff
@@ -229,6 +229,7 @@ function createRoot() {
     const root = ReactDOMClient.createRoot(document.documentElement);
 
     root.render(<div>initial render</div>);
+    root.render(false);
```
Comment (eps1lon) [inline types/react-dom/test/react-dom-tests.tsx:232]: Another kind of node: https://codesandbox.io/s/initialchildrne-accepts-reactnode-exy9jm

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-dom ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-dom@latest/))
was missing the following properties:
1. createRoot
1. hydrateRoot
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2c703ee5344151c649b3ef7022da7b461c584ecd
</p>

**Pull request merged**